website/content/docs/secrets/pki/rotation-primitives.mdx

```diff
@@ -0,0 +1,424 @@
+connecting to the service would not know the difference.
+
+Rotation of intermediate certificates is almost as easy. Assuming a decent
+operational setup (wherein during end-entity issuance, the full certificate
+chain is updated in the service's configuration), this should be as easy as
+creating a new intermediate CA, signing it against the root CA, and then
+beginning issuance against the new intermediate certificate. In Vault, if
+the intermediate is generated in an existing mount path (or is moved into
+such), the requesting entity shouldn't care much. Under ACME, Let's Encrypt
+has successfully rotated intermediates to present a cross-signed chain
+([for older Android devices](https://letsencrypt.org/2020/12/21/extending-android-compatibility.html)).
+Assuming the old intermediate's parent(s) are still valid and trusted,
+certificates issued under old intermediates should continue to validate.
+
+The hard part of rotation--calling for the use of these primitives--is
+rotating root certificates. These live in every device's trust store and
+are hard to update from an organization-wide operational perspective.
+Unless the organization can swap out roots almost instantaneously and
+simultaneously (e.g., via an agent) with no missed devices, this process
+will likely span months.
+
+To make this process lower risk, there are various primitive certificate
+types that use the [above certificate fields](#x-509-certificate-fields).
+Key to their success is the following note:
+
+~> Note: While certificates are added to the trust store, it is ultimately
+   the associated key material that determines trust: two issuer certificates
+   with the same subject but different public keys cannot validate the same
+   leaf certificate; only if the keys are the same can this occur.
+
+### Cross-Signed Primitive
+
+This is the most common type of rotation primitive. A common CSR is signed by
+two CAs, resulting in two certificates. These certificates must have the same
+Subject (but may have different Issuers and will have different Serial Numbers)
+and the same backing key material, to allow certificates they sign to be
+trusted by either variant.
+
+Note that, due to restrictions in how end-entity certificates are used and
+validated (services and validation libraries expect only one), cross-signing
+most typically only applies to intermediate.
+
+#### Cross-Signed Roots
+
+Technically, cross-signing can occur between two roots, allowing trust bundles
+with either root to validate certs issued through the other. However, this
+process creates a certificate that is effectively an intermediate (as it is
+no longer self-signed) and usually must be served alongside the trust chain.
+Given this restriction, it's preferable to instead cross-sign the top-level
+intermediates under the root unless strictly necessary when the old root
+certificate has been used to directly issue leaf certificates.
+
+##### Process Flow
+
+```
+        -------------------
```

Comment:
I wonder if we can get education to help us with some art.

Reply:
Yeah... :-D